docs/Recovery-Codes.md

```diff
@@ -0,0 +1,99 @@
+# Recovery Codes
+
+The recovery codes allow your users to recover their activation in case that mobile device is lost or stolen. Before you start, please read the [Activation Recovery](https://github.com/wultra/powerauth-crypto/blob/develop/docs/Activation-Recovery.md) document, available in our [powerauth-crypto](https://github.com/wultra/powerauth-crypto) repository.
+
+## Recovery Code Data
+
+To recover an activation, the user has to re-type two separate values:
+
+1. **Recovery Code** itself, which is very similar to an activation code. So you can detect typing errors before you submit such code to the server.
+2. **PUK**, which is an additional numeric value and acts as a one-time password in the scheme.
+
+## Recovery Code Scenarios
+
+PowerAuth currently supports two basic types of recovery codes:
+
+1. Recovery Code bound to a previous PowerAuth activation.
+   - This type of code can be obtained only in an already activated application.
+   - This type of code has only one PUK available, so only one recovery operation is possible.
+   - The activation associated with the code is removed once the recovery operation succeeds.
+
+2. Recovery Code delivered via OOB channel, typically in the form of a securely printed postcard, delivered by the post service.
+   - This type of code has typically more than one PUK associated with the code, so it can be used multiple times.
+   - The user has to keep that postcard in a safe and secure place, and mark already used PUKs.
+   - The code delivery must be confirmed by the user before the code can be used for a recovery operation.
+
+The feature is not automatically available. It must be enabled and configured on PowerAuth Server. If it's so, then your mobile application can use several methods related to this feature.
+
+## Getting Recovery Data
+
+If the recovery data was received during the activation process, then you can later display that information to the user. To check the existence of recovery data and get that information, use the following code:
+
+```javascript
+const hasRecovery = await PowerAuth.hasActivationRecoveryData();
+if (!hasRecovery) {
+    // Recovery information is not available
+    return;
+}
+
+// 2FA signature, uses device related key and user PIN code
+const auth = new PowerAuthAuthentication();
+auth.usePossession = true;
+auth.userPassword = "1234";
+
+try {
+    const recoveryData = await PowerAuth.activationRecoveryData(auth);
+    const recoveryCode = recoveryData.recoveryCode;
+    const puk = recoveryData.puk;
+    // Show values on the screen
+} catch (e) {
+    // Show an error
+}
+```
+
+<!-- begin box warning -->
```

Comment:
Ouch... this is incorrect and probably a result of a partial pull from the SDK main repo, where I had this issue. This information should not be in the BOX macro.